### PR TITLE
Task 12 Delete

### PR DESCRIPTION
**Pull request by Hashim-47**
12. DELETE /api/comments/:comment_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -409,3 +409,17 @@ describe("GET /api/users", () => {
...
   });
 });
+
+describe("DELETE /api/comments/:comment_id", () => {
+  test("respond with code 204 and delete the given comment by comment_id", () => {
+    return request(app)
+      .delete("/api/comments/1")
+      .expect(204)
+      .then(() => {
+        return db.query("SELECT * FROM comments WHERE comment_id = 1");
```
Comment (Rosem10) [inline __tests__/app.test.js]: It would be far better to use your getCommentsById endpoint here! Nice thinking though

File: app.js [hunk 2]
```diff
@@ -41,6 +43,7 @@ app.use((err, req, res, next) => {
 });
 
 app.use((err, req, res, next) => {
+  console.log(err);
```
Comment (Rosem10) [inline app.js]: Remove please!

File: __tests__/app.test.js (hunk 1 above)
Comment (Rosem10) [reply __tests__/app.test.js]: What if they try to delete a comment that doesn't exist, or one with an invalid id?

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const { deleteCommentById } = require("../models/comments.model");
+
+const deleteComment = (req, res, next) => {
+  const { comment_id } = req.params;
+
+  deleteCommentById(comment_id)
+    .then(() => {
+      res.status(204).send();
```
Comment (Rosem10) [inline controllers/comments.controller.js]: There's another option here to use res.sendStatus(204) - which will do both status and send in one go.

File: models/comments.model.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const db = require("../db/connection");
+
+const deleteCommentById = (comment_id) => {
+  return db
+    .query("DELETE FROM comments WHERE comment_id = $1 RETURNING * ", [
+      comment_id,
+    ])
+    .then((res) => {
+      if (res.rows.length === 0) {
+        console.log(res.rows);
```
Comment (Rosem10) [inline models/comments.model.js]: Remove this too, please

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Take a look at some error handling please!

**Review by MKD01 — APPROVED** [review]
Looks good, happy for this to be merged

**Pull request merged**